### ISSUE TITLE
build seem to fail on MacOS

### ISSUE THREAD

**Issue by edibleparts** [yurikoles/homebrew-yurikoles#3]
macOS Catalina, latest XCode command-line tools installed.
tried both
brew install yurikoles/yurikoles/far2l
and
brew install yurikoles/yurikoles/far2l --HEAD
Build goes on for a while, then fails on maltiarc build. E.g.
==> Installing far2l from yurikoles/yurikoles
==> Cloning https://github.com/elfmz/far2l.git
Cloning into '/Users/z/Library/Caches/Homebrew/far2l--git'...
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
==> cmake .
==> make install
Last 15 lines from /Users/z/Library/Logs/Homebrew/far2l/02.make:
cd /tmp/far2l-20200831-67634-1jp0b1h/multiarc && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -DHAVE_PCRE -DRARDLL -DWXUSINGDLL -D_7ZIP_PPMD_SUPPPORT -D_7ZIP_ST -D_FILE_OFFSET_BITS=64 -D__WXMAC__ -D__WXOSX_COCOA__ -D__WXOSX__ -Dmultiarc_EXPORTS -I/tmp/far2l-20200831-67634-1jp0b1h/utils/include -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/formats/rar/unrar -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/libpcre -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../WinPort -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../far2l -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../far2l/Include -isystem /usr/local/lib/wx/include/osx_cocoa-unicode-3.0 -isystem /usr/local/include/wx-3.0 -Wall -std=c99 -fPIC -Wno-unused-function -D_FILE_OFFSET_BITS=64  -Wall -std=c99 -fPIC -Wno-unused-function -D_FILE_OFFSET_BITS=64 -O2 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -fPIC -fvisibility=hidden -o CMakeFiles/multiarc.dir/src/formats/7z/C/Bra.c.o -c /tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/formats/7z/C/Bra.c
[ 83%] Building C object multiarc/CMakeFiles/multiarc.dir/src/formats/7z/C/Blake2s.c.o
cd /tmp/far2l-20200831-67634-1jp0b1h/multiarc && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -DHAVE_PCRE -DRARDLL -DWXUSINGDLL -D_7ZIP_PPMD_SUPPPORT -D_7ZIP_ST -D_FILE_OFFSET_BITS=64 -D__WXMAC__ -D__WXOSX_COCOA__ -D__WXOSX__ -Dmultiarc_EXPORTS -I/tmp/far2l-20200831-67634-1jp0b1h/utils/include -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/formats/rar/unrar -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/libpcre -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../WinPort -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../far2l -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../far2l/Include -isystem /usr/local/lib/wx/include/osx_cocoa-unicode-3.0 -isystem /usr/local/include/wx-3.0 -Wall -std=c99 -fPIC -Wno-unused-function -D_FILE_OFFSET_BITS=64  -Wall -std=c99 -fPIC -Wno-unused-function -D_FILE_OFFSET_BITS=64 -O2 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -fPIC -fvisibility=hidden -o CMakeFiles/multiarc.dir/src/formats/7z/C/Blake2s.c.o -c /tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/formats/7z/C/Blake2s.c
[ 83%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/custom/custom.cpp.o
cd /tmp/far2l-20200831-67634-1jp0b1h/multiarc && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -DHAVE_PCRE -DRARDLL -DWXUSINGDLL -D_7ZIP_PPMD_SUPPPORT -D_7ZIP_ST -D_FILE_OFFSET_BITS=64 -D__WXMAC__ -D__WXOSX_COCOA__ -D__WXOSX__ -Dmultiarc_EXPORTS -I/tmp/far2l-20200831-67634-1jp0b1h/utils/include -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/formats/rar/unrar -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/libpcre -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../WinPort -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../far2l -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../far2l/Include -isystem /usr/local/lib/wx/include/osx_cocoa-unicode-3.0 -isystem /usr/local/include/wx-3.0 -Wall -fPIC -Wno-unused-function -Wno-c++11-narrowing -D_FILE_OFFSET_BITS=64  -Wall -fPIC -Wno-unused-function -Wno-c++11-narrowing -D_FILE_OFFSET_BITS=64 -O2 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -fPIC -fvisibility=hidden -std=c++11 -o CMakeFiles/multiarc.dir/src/formats/custom/custom.cpp.o -c /tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/formats/custom/custom.cpp
[ 83%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/targz/targz.cpp.o
cd /tmp/far2l-20200831-67634-1jp0b1h/multiarc && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -DHAVE_PCRE -DRARDLL -DWXUSINGDLL -D_7ZIP_PPMD_SUPPPORT -D_7ZIP_ST -D_FILE_OFFSET_BITS=64 -D__WXMAC__ -D__WXOSX_COCOA__ -D__WXOSX__ -Dmultiarc_EXPORTS -I/tmp/far2l-20200831-67634-1jp0b1h/utils/include -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/formats/rar/unrar -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/libpcre -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../WinPort -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../far2l -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../far2l/Include -isystem /usr/local/lib/wx/include/osx_cocoa-unicode-3.0 -isystem /usr/local/include/wx-3.0 -Wall -fPIC -Wno-unused-function -Wno-c++11-narrowing -D_FILE_OFFSET_BITS=64  -Wall -fPIC -Wno-unused-function -Wno-c++11-narrowing -D_FILE_OFFSET_BITS=64 -O2 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -fPIC -fvisibility=hidden -std=c++11 -o CMakeFiles/multiarc.dir/src/formats/targz/targz.cpp.o -c /tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/formats/targz/targz.cpp
[ 83%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/cab/cab.cpp.o
cd /tmp/far2l-20200831-67634-1jp0b1h/multiarc && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -DHAVE_PCRE -DRARDLL -DWXUSINGDLL -D_7ZIP_PPMD_SUPPPORT -D_7ZIP_ST -D_FILE_OFFSET_BITS=64 -D__WXMAC__ -D__WXOSX_COCOA__ -D__WXOSX__ -Dmultiarc_EXPORTS -I/tmp/far2l-20200831-67634-1jp0b1h/utils/include -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/formats/rar/unrar -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/libpcre -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../WinPort -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../far2l -I/tmp/far2l-20200831-67634-1jp0b1h/multiarc/../far2l/Include -isystem /usr/local/lib/wx/include/osx_cocoa-unicode-3.0 -isystem /usr/local/include/wx-3.0 -Wall -fPIC -Wno-unused-function -Wno-c++11-narrowing -D_FILE_OFFSET_BITS=64  -Wall -fPIC -Wno-unused-function -Wno-c++11-narrowing -D_FILE_OFFSET_BITS=64 -O2 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -fPIC -fvisibility=hidden -std=c++11 -o CMakeFiles/multiarc.dir/src/formats/cab/cab.cpp.o -c /tmp/far2l-20200831-67634-1jp0b1h/multiarc/src/formats/cab/cab.cpp
[ 83%] Linking CXX shared module ../install/Plugins/multiarc/plug/multiarc.far-plug-mb
cd /tmp/far2l-20200831-67634-1jp0b1h/multiarc && /usr/local/Cellar/cmake/3.18.2/bin/cmake -E cmake_link_script CMakeFiles/multiarc.dir/link.txt --verbose=1
/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++  -Wall -fPIC -Wno-unused-function -Wno-c++11-narrowing -D_FILE_OFFSET_BITS=64  -Wall -fPIC -Wno-unused-function -Wno-c++11-narrowing -D_FILE_OFFSET_BITS=64 -O2 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -bundle -Wl,-headerpad_max_install_names -flat_namespace -undefined suppress -o ../install/Plugins/multiarc/plug/multiarc.far-plug-mb CMakeFiles/multiarc.dir/src/MultiArc.cpp.o CMakeFiles/multiarc.dir/src/ArcPlg.cpp.o CMakeFiles/multiarc.dir/src/arccfg.cpp.o CMakeFiles/multiarc.dir/src/arcget.cpp.o CMakeFiles/multiarc.dir/src/arcput.cpp.o CMakeFiles/multiarc.dir/src/arcreg.cpp.o CMakeFiles/multiarc.dir/src/ArcMix.cpp.o CMakeFiles/multiarc.dir/src/ArcProc.cpp.o CMakeFiles/multiarc.dir/src/global.cpp.o CMakeFiles/multiarc.dir/src/arcread.cpp.o CMakeFiles/multiarc.dir/src/arccmd.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/rar.cpp.o CMakeFiles/multiarc.dir/src/formats/ha/ha.cpp.o CMakeFiles/multiarc.dir/src/formats/arj/arj.cpp.o CMakeFiles/multiarc.dir/src/formats/ace/ace.cpp.o CMakeFiles/multiarc.dir/src/formats/arc/arc.cpp.o CMakeFiles/multiarc.dir/src/formats/zip/zip.cpp.o CMakeFiles/multiarc.dir/src/formats/lzh/lzh.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/scantree.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/sha1.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/threadpool.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/crc.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/hash.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/strlist.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/rs16.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/resource.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/consio.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/secpassword.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/qopen.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/recvol.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/rijndael.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/timefn.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/encname.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/rawread.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/file.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/strfn.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/sha256.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/filefn.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/filcreat.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/headers.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/blake2s.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/options.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/volume.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/isnt.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/system.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/crypt.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/rar.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/cmddata.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/extinfo.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/filestr.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/ui.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/list.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/find.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/smallfn.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/unicode.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/pathfn.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/global.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/rarvm.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/getbits.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/rs.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/errhnd.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/archive.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/dll.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/extract.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/match.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/unpack.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/arcread.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/rdwrfn.cpp.o CMakeFiles/multiarc.dir/src/formats/rar/unrar/rarpch.cpp.o CMakeFiles/multiarc.dir/src/formats/7z/7z.cpp.o CMakeFiles/multiarc.dir/src/formats/7z/7zMain.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Lzma86Dec.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/7zDec.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Ppmd7Dec.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Sha1.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/BraIA64.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Xz.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Lzma2Dec.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/AesOpt.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/BwtSort.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Bra86.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/7zCrcOpt.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/7zAlloc.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/LzmaLib.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/HuffEnc.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/7zBuf.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/XzDec.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/7zBuf2.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Lzma86Enc.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Alloc.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/XzCrc64.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/XzEnc.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Ppmd8.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Aes.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/7zCrc.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Bcj2.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Delta.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/7zFile.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Bcj2Enc.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Ppmd8Enc.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/LzmaEnc.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Ppmd7Enc.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/7zStream.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Ppmd8Dec.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/XzCrc64Opt.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/LzmaDec.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/XzIn.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/CpuArch.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/LzFind.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Ppmd7.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Lzma2Enc.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Sha256.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Sort.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/7zArcIn.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Bra.c.o CMakeFiles/multiarc.dir/src/formats/7z/C/Blake2s.c.o CMakeFiles/multiarc.dir/src/formats/custom/custom.cpp.o CMakeFiles/multiarc.dir/src/formats/targz/targz.cpp.o CMakeFiles/multiarc.dir/src/formats/cab/cab.cpp.o  ../utils/libutils.a -Wl,-bundle_loader,/tmp/far2l-20200831-67634-1jp0b1h/install/far2l /usr/local/lib/libpcre.dylib 
cd /tmp/far2l-20200831-67634-1jp0b1h/multiarc && /usr/local/Cellar/cmake/3.18.2/bin/cmake -E copy_directory /tmp/far2l-20200831-67634-1jp0b1h/multiarc/configs /tmp/far2l-20200831-67634-1jp0b1h/install/Plugins/multiarc
[ 83%] Built target multiarc
make: *** [all] Error 2

Any ideas how to proceed?  Thanks a lot!

**Comment by edibleparts** [comment]
Logging.cpp does not compile:

/Users/z/dev/far2l/colorer/src/shared/common/Logging.cpp:13:14: error: redefinition
      of 'log' as different kind of symbol
static FILE *log = 0;
             ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:1003:1: note: previous
      definition is here
log(_A1 __lcpp_x) _NOEXCEPT {return ::log((double)__lcpp_x);}
^
/Users/z/dev/far2l/colorer/src/shared/common/Logging.cpp:19:10: error: reference to
      overloaded function could not be resolved; did you mean to call it?
  while (log == 0 && idx < 10){
         ^~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:1003:1: note: possible
      target for call
log(_A1 __lcpp_x) _NOEXCEPT {return ::log((double)__lcpp_x);}
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:997:46: note: possible
      target for call
inline _LIBCPP_INLINE_VISIBILITY long double log(long double __lcpp_x) _NOEXCEPT {re...
                                             ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:996:46: note: possible
      target for call
inline _LIBCPP_INLINE_VISIBILITY float       log(float __lcpp_x) _NOEXCEPT       {re...
                                             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/math.h:373:15: note: 
      possible target for call
extern double log(double);
              ^
/Users/z/dev/far2l/colorer/src/shared/common/Logging.cpp:26:5: error: reference to
      overloaded function could not be resolved; did you mean to call it?
    log = fopen(log_name, "ab+");
    ^~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:1003:1: note: possible
      target for call
log(_A1 __lcpp_x) _NOEXCEPT {return ::log((double)__lcpp_x);}
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:997:46: note: possible
      target for call
inline _LIBCPP_INLINE_VISIBILITY long double log(long double __lcpp_x) _NOEXCEPT {re...
                                             ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:996:46: note: possible
      target for call
inline _LIBCPP_INLINE_VISIBILITY float       log(float __lcpp_x) _NOEXCEPT       {re...
                                             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/math.h:373:15: note: 
      possible target for call
extern double log(double);
              ^
/Users/z/dev/far2l/colorer/src/shared/common/Logging.cpp:30:3: error: no matching
      function for call to 'fprintf'
  fprintf(log, "[%s][%s] ", levelNames[level], cname);
  ^~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdio.h:155:6: note: 
      candidate function not viable: no overload of 'log' matching 'FILE *'
      (aka '__sFILE *') for 1st argument
int      fprintf(FILE * __restrict, const char * __restrict, ...) __printflike(2, 3);
         ^
/Users/z/dev/far2l/colorer/src/shared/common/Logging.cpp:32:3: error: no matching
      function for call to 'vfprintf'
  vfprintf(log, msg, v);
  ^~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdio.h:190:6: note: 
      candidate function not viable: no overload of 'log' matching 'FILE *'
      (aka '__sFILE *') for 1st argument
int      vfprintf(FILE * __restrict, const char * __restrict, va_list) __printflike(2, 0);
         ^
/Users/z/dev/far2l/colorer/src/shared/common/Logging.cpp:34:3: error: no matching
      function for call to 'fprintf'
  fprintf(log, "\n");
  ^~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdio.h:155:6: note: 
      candidate function not viable: no overload of 'log' matching 'FILE *'
      (aka '__sFILE *') for 1st argument
int      fprintf(FILE * __restrict, const char * __restrict, ...) __printflike(2, 3);
         ^
/Users/z/dev/far2l/colorer/src/shared/common/Logging.cpp:36:3: error: no matching
      function for call to 'fflush'
  fflush(log);
  ^~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdio.h:146:6: note: 
      candidate function not viable: no overload of 'log' matching 'FILE *'
      (aka '__sFILE *') for 1st argument
int      fflush(FILE *);
         ^
/Users/z/dev/far2l/colorer/src/shared/common/Logging.cpp:51:7: error: reference to
      overloaded function could not be resolved; did you mean to call it?
  if (log != 0){
      ^~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:1003:1: note: possible
      target for call
log(_A1 __lcpp_x) _NOEXCEPT {return ::log((double)__lcpp_x);}
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:997:46: note: possible
      target for call
inline _LIBCPP_INLINE_VISIBILITY long double log(long double __lcpp_x) _NOEXCEPT {re...
                                             ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:996:46: note: possible
      target for call
inline _LIBCPP_INLINE_VISIBILITY float       log(float __lcpp_x) _NOEXCEPT       {re...
                                             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/math.h:373:15: note: 
      possible target for call
extern double log(double);
              ^
/Users/z/dev/far2l/colorer/src/shared/common/Logging.cpp:52:5: error: no matching
      function for call to 'fclose'
    fclose(log);
    ^~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdio.h:143:6: note: 
      candidate function not viable: no overload of 'log' matching 'FILE *'
      (aka '__sFILE *') for 1st argument
int      fclose(FILE *);
         ^
/Users/z/dev/far2l/colorer/src/shared/common/Logging.cpp:54:3: error: reference to
      overloaded function could not be resolved; did you mean to call it?
  log = fopen(logfile, "ab+");
  ^~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:1003:1: note: possible
      target for call
log(_A1 __lcpp_x) _NOEXCEPT {return ::log((double)__lcpp_x);}
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:997:46: note: possible
      target for call
inline _LIBCPP_INLINE_VISIBILITY long double log(long double __lcpp_x) _NOEXCEPT {re...
                                             ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:996:46: note: possible
      target for call
inline _LIBCPP_INLINE_VISIBILITY float       log(float __lcpp_x) _NOEXCEPT       {re...
                                             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/math.h:373:15: note: 
      possible target for call
extern double log(double);
              ^

**Comment by ariel777** [comment]
Similar issue:
cd /tmp/far2l-20200903-13123-1ay7mkj/far2l-alpha-23jun19/multiarc && /usr/local/Cellar/cmake/3.18.2/bin/cmake -E copy_directory /tmp/far2l-20200903-13123-1ay7mkj/far2l-alpha-23jun19/multiarc/configs /tmp/far2l-20200903-13123-1ay7mkj/far2l-alpha-23jun19/install/Plugins/multiarc
[ 49%] Built target multiarc
make: *** [all] Error 2

**Comment by Bregor** [comment]
```==> cmake .. -G Ninja
Last 15 lines from /Users/bregor/Library/Logs/Homebrew/far2l/01.cmake:
  VERSION keyword not followed by a value or was followed by a value that
  expanded to nothing.


-- Could NOT find MiniZip (missing: MINIZIP_LIBRARY MINIZIP_INCLUDE_DIR)
CMake Error at /usr/local/Cellar/cmake/3.18.3/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find Spdlog (missing: SPDLOG_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.18.3/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindSpdlog.cmake:24 (find_package_handle_standard_args)
  colorer/src/Colorer-library/src/CMakeLists.txt:173 (FIND_PACKAGE)``

**Comment by graycrow** [comment]
> -- Could NOT find MiniZip (missing: MINIZIP_LIBRARY MINIZIP_INCLUDE_DIR)

I was able to continue the build by installing MiniZip by `brew install minizip`
Then it failed on SpdLog, fixed by `brew install spdlog`
And then finally failed on 
```
-- Configuring done
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
/tmp/far2l-20201004-71741-12f35kn/colorer/XERCESC_INCLUDE_DIR
   used as include directory in directory /tmp/far2l-20201004-71741-12f35kn/colorer
/tmp/far2l-20201004-71741-12f35kn/colorer/src/Colorer-library/src/XERCESC_INCLUDE_DIR
   used as include directory in directory /tmp/far2l-20201004-71741-12f35kn/colorer/src/Colorer-library/src

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

macOS 10.14.6 with all updates

UPDATE: `brew install xerces-c` helped. Build succeeded.

**Comment by graycrow** [comment]
To summarise, if build fails with `-- Could NOT find MiniZip (missing: MINIZIP_LIBRARY MINIZIP_INCLUDE_DIR)` then installing the following 3 packages may help. This was necessary for my system, your one might be different, so install them one by one and try to build after each one. 
```
brew install minizip
brew install spdlog
brew install xerces-c
```

**Comment by Bregor** [comment]
@graycrow thanks!
Built successfully at last :)

**Comment by snowytoxa** [comment]
same for me, but I also had to install `uchardet`
so I made a PR to address this https://github.com/yurikoles/homebrew-yurikoles/pull/6

**Comment by yurikoles** [comment]
Should be fixed in 46682a6f8156f90c2473e35cf3d3e68b172675c0
